client/js/widgets/button.js

```diff
@@ -69,10 +69,20 @@ export class Button extends Widget {
       if(a.applyVariables) {
         if(Array.isArray(a.applyVariables)) {
           for(const v of a.applyVariables) {
-            if(v.parameter && v.variable)
-              a[v.parameter] = variables[v.variable];
-            else
-              problems.push('Entry in parameter applyVariables does not contain "parameter" and "variable".');
+            if(v.parameter && v.variable) {
+              a[v.parameter] = (v.index === undefined) ? variables[v.variable] : variables[v.variable][v.index];
+            } else if (v.parameter && v.template) {
+              for (let key in variables) {
```

Comment:
Does this `for` loop do anything? It looks like you read the `key` from the RegEx match two lines further down.